src/backend/distributed/planner/multi_explain.c

```diff
@@ -199,6 +201,20 @@ CitusExplainScan(CustomScanState *node, List *ancestors, struct ExplainState *es
 		return;
 	}
 
+	if (SkipConstraintValidation || (AlterTableInProgress() && IsCitusInternalBackend()))
+	{
+		/*
+		 * We do not want to explain ALTER TABLE ... ADD FOREIGN KEY .. queries in
+		 * in the workers. They trigger a SELECT query on the citus tables as part of the
+		 * constraint validation check. Explaining the distributed SELECT query in the
+		 * workers cause memory corruption resulting in a crash later on.
+		 */
+		appendStringInfoSpaces(es->str, es->indent * 2);
+		appendStringInfo(es->str, "explain statements for distributed queries ");
+		appendStringInfo(es->str, "triggered via DDL commands are only supported on the coordinator \n");
```

Comment:
I wonder we should at least add a single test:
```SQL
LOAD 'auto_explain';
SET auto_explain.log_min_duration = 0;
SET auto_explain.log_level = LOG;
SET client_min_messages to LOG;
SET auto_explain.log_timing TO off;

BEGIN;
   -- simulate being a worker session/backend
	SET LOCAL application_name to 'citus_internal gpid=10000000001';
	SET citus.enable_ddl_propagation TO OFF;
    -- alter table triggers SELECT, and auto_explain catches that
	ALTER TABLE target_table ADD CONSTRAINT fkey_167 FOREIGN KEY (col_1) REFERENCES test_ref_table(key) ON DELETE CASCADE;
	LOG:  duration: 0.008 ms  plan:
Query Text: SELECT fk."col_1" FROM ONLY "public"."target_table" fk LEFT OUTER JOIN ONLY "public"."test_ref_table" pk ON ( pk."key" OPERATOR(pg_catalog.=) fk."col_1") WHERE pk."key" IS NULL AND (fk."col_1" IS NOT NULL)
Custom Scan (Citus Adaptive)  (cost=0.00..0.00 rows=100000 width=4) (never executed)
  explain statements for distributed queries triggered via DDL commands are only supported on the coordinator 
ALTER TABLE
```

But, we should find a way to make the test outputs consistent, because we somehow have `LOG:  **duration: 0.008** ms  plan:` in the output, which would change across executions.

Reply:
friendly reminder: I think it'd be much better to have at least one test covering this case, the above is an example/suggestion, but anything is welcome

Reply:
It looks like we can normalize the variable data in the output.